### PR TITLE
fix bug in lstm model

### PR DESCRIPTION
**Pull request by Zrachel**
The lstm model losses a projection from lstm hidden state to out_embed. Is it a bug?

### CODE REVIEW

**Comment by myleott** [comment]
Good catch! We were trying to follow the implementation here [1], which doesn’t have the extra projection. But as you noticed, the current fairseq version doesn’t work if out_embed_dim != embed_dim. How about if we conditionally apply the projection only when out_embed_dim != embed_dim?

[1] https://github.com/harvardnlp/seq2seq-attn/

File: fairseq/models/lstm.py [hunk 1]
```diff
@@ -197,6 +199,9 @@ def _forward(self, input_tokens, encoder_out):
         attn_scores = attn_scores.transpose(0, 2)
 
         # project back to size of vocabulary
+        if hasattr(self, 'additional_fc'):
+            x = self.additional_fc(x)
+            x = F.dropout(x, p=self.dropout, training=self.training)
```
Comment (myleott) [inline fairseq/models/lstm.py]: I think we want `p=self.dropout_out` here.

Comment (Zrachel) [reply fairseq/models/lstm.py]: Sorry, I forgot I've changed to one dropout in my own code.
